Index: new/gfs2/init.d/gfs2-cluster
===================================================================
--- new.orig/gfs2/init.d/gfs2-cluster	2012-02-17 09:01:54.000000000 +0100
+++ new/gfs2/init.d/gfs2-cluster	2012-02-17 10:09:57.000000000 +0100
@@ -19,27 +19,28 @@
 # Description: Starts and stops gfs_controld for dlm based GFS2 filesystems
 ### END INIT INFO
 
-. /etc/rc.d/init.d/functions
+. /lib/lsb/init-functions
 
 prog="gfs_controld"
-lockfile="/var/lock/subsys/$prog"
+lockfile="/var/lock/$prog"
 exec="/usr/sbin/$prog"
+DESC="GFS2 Control Daemon"
 
 start() {
 	[ -x $exec ] || exit 5
-	echo -n $"Starting $prog: "
-	daemon $prog
+	log_daemon_msg "Starting $DESC" "$prog"
+	start_daemon $exec
 	retval=$?
-	echo
+	log_end_msg $retval
 	[ $retval -eq 0 ] && touch $lockfile
 	return $retval
 }
 
 stop() {
-	echo -n $"Stopping $prog: "
+	log_daemon_msg "Stopping $DESC" "$prog"
 	killproc $prog
 	retval=$?
-	echo
+	log_end_msg $retval
 	[ $retval -eq 0 ] && rm -f $lockfile
 }
 
@@ -53,7 +54,7 @@
 }
 
 rh_status() {
-	status $prog
+        status_of_proc $exec $prog
 }
 
 rh_status_q() {
@@ -87,7 +88,7 @@
 		restart
 		;;
 	*)
-		echo $"Usage $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
+		echo "Usage $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
 		exit 2
 esac
 exit $?
